contracts/pair_anchor/src/contract.rs

```diff
@@ -0,0 +1,860 @@
+            } else {
+                None
+            };
+
+            swap(
+                deps,
+                env,
+                info,
+                Addr::unchecked(cw20_msg.sender),
+                Asset {
+                    info: AssetInfo::Token { contract_addr },
+                    amount: cw20_msg.amount,
+                },
+                belief_price,
+                max_spread,
+                to_addr,
+            )
+        }
+        Ok(Cw20HookMsg::WithdrawLiquidity {}) => Err(ContractError::NonSupported {}),
+        Err(err) => Err(ContractError::Std(err)),
+    }
+}
+
+/// ## Description
+/// Performs an swap operation with the specified parameters. The trader must approve the
+/// pool contract to transfer offer assets from their wallet.
+/// Returns an [`ContractError`] on failure, otherwise returns the [`Response`] with the specified attributes if the operation was successful.
+/// ## Params
+/// * **deps** is an object of type [`DepsMut`].
+///
+/// * **env** is an object of type [`Env`].
+///
+/// * **info** is an object of type [`MessageInfo`].
+///
+/// * **sender** is an object of type [`Addr`]. This is the sender of the swap operation.
+///
+/// * **offer_asset** is an object of type [`Asset`]. Proposed asset for swapping.
+///
+/// * **belief_price** is an object of type [`Option<Decimal>`]. Used to calculate the maximum swap spread.
+///
+/// * **max_spread** is an object of type [`Option<Decimal>`]. Sets the maximum spread of the swap operation.
+///
+/// * **to** is an object of type [`Option<Addr>`]. Sets the recipient of the swap operation.
+/// NOTE - the address that wants to swap should approve the pair contract to pull the offer token.
+#[allow(clippy::too_many_arguments)]
+pub fn swap(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    sender: Addr,
+    offer_asset: Asset,
+    belief_price: Option<Decimal>,
+    max_spread: Option<Decimal>,
+    to: Option<Addr>,
+) -> Result<Response, ContractError> {
+    offer_asset.assert_sent_native_token_balance(&info)?;
+
+    let config: Config = CONFIG.load(deps.storage)?;
+
+    // If the asset balance is already increased, we should subtract the user deposit from the pool amount
+    let pools: Vec<Asset> = config
+        .pair_info
+        .query_pools(&deps.querier, env.clone().contract.address)?
+        .iter()
+        .map(|p| {
+            let p = p.clone();
+            p
+        })
+        .collect();
+
+    let offer_pool: Asset;
+    let ask_pool: Asset;
+
+    if offer_asset.info.equal(&pools[0].info) {
+        offer_pool = pools[0].clone();
+        ask_pool = pools[1].clone();
+    } else if offer_asset.info.equal(&pools[1].info) {
+        offer_pool = pools[1].clone();
+        ask_pool = pools[0].clone();
+    } else {
+        return Err(ContractError::AssetMismatch {});
+    }
+
+    if ask_pool.amount > Uint128::from(0u128) {
+        // send to community? Otherwise it would be sent to the swapper
```

Comment:
when executing a swap, but someone manually sent funds (aust) to the contract address, currently the funds will be transferred to the recipient of the current swap, as AssertAndSend will just send all available tokens to the recipient.

Reply:
Let's send it to the maker contract

Reply:
done

Reply:
done